[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
fastapi = ">=0.63.0,<1"
uvicorn = {version = ">=0.13.3,<1", extras = ["standard"]}
jinja2 = ">=2.11.3,<3"
aiofiles = ">=0.6.0,<1"
python-multipart = ">=0.0.5,<1"
pydantic = {extras = ["email"], version = ">=1.8.1,<2"}
alembic = ">=1.5.8,<2"
gino = ">=1.0.1,<2"
starlette = ">=0.13.6,<1"
psycopg2-binary = ">=2.8.6,<3"
argon2-cffi = ">=20.1.0,<21"
pyjwt = ">=2.0.1,<3"
sqlalchemy = ">=1.2,<2"
expiringdict = ">=1.2.1,<2"
fido2 = ">=0.9.1,<1"

[dev-packages]
black = "*"
flake8 = "*"
isort = "*"
pre-commit = "*"

[requires]
python_version = "3.9"

[pipenv]
allow_prereleases = true

[scripts]
web = "uvicorn web.main:app --reload"
# The fido2 library only verifies origins with http*s* in the url. Therefore you have to
# use an https connection. For testing you can just generate a self signed certificate
# with the command below and then run ``pipenv run web_secure``
# openssl req -x509 -out localhost.crt -keyout localhost.key \                                                                                                                                                                     ─╯
#  -newkey rsa:2048 -nodes -sha256 \
#  -subj '/CN=localhost' -extensions EXT -config <( \
#   printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
web_secure = "uvicorn web.main:app --host localhost --reload --ssl-keyfile localhost.key --ssl-certfile localhost.crt"
lint = "pre-commit run -all-files"
precommit = "pre-commit install"
migrations = "alembic upgrade head"
